src/cider/tests/nextgen/functional/CiderMathOpNextgenTest.cpp

```diff
@@ -0,0 +1,381 @@
+  assertQueryArrow(
+      "SELECT integer_col * bigint_col,  bigint_col / integer_col FROM test where "
+      "integer_col <> 0");
+
+  assertQueryArrow(
+      "SELECT bigint_col * double_col + integer_col * float_col FROM test where "
+      "bigint_col * double_col > integer_col * "
+      "float_col ");
+
+  assertQueryArrow(
+      "SELECT bigint_col * double_col - integer_col * float_col FROM test where "
+      "bigint_col * double_col > integer_col * "
+      "float_col ");
+
+  assertQueryArrow(
+      "SELECT (double_col - float_col) * (integer_col - bigint_col) FROM test where "
+      "double_col > float_col and integer_col "
+      "> bigint_col");
+
+  assertQueryArrow(
+      "SELECT (double_col - float_col) / (integer_col - bigint_col) FROM test where "
+      "double_col > float_col and integer_col "
+      "> bigint_col");
+
+  assertQueryArrow(
+      "SELECT double_col + integer_col / bigint_col FROM test where bigint_col <> 0");
+
+  // Using modified substrait-java generated json file by version v0.7.0 format to pass
+  // test as isthmus will cast double into float, loss of precision in float op double
+  // case.
+  assertQueryArrow("SELECT float_col + double_col FROM test",
+                   "mathop_add_fp32_fp64.json");
+}
+
+TEST_F(CiderMathOpArrowTest, ConstantMathOpTest) {
+  assertQueryArrow("SELECT double_col + 1.23e1, double_col - 1.23e1 FROM test");
+  assertQueryArrow("SELECT bigint_col * 1.23e1,  bigint_col / 1.23e1 FROM test");
+  assertQueryArrow("SELECT 3 * (1.23e1 + bigint_col) / 2 FROM test");
+  assertQueryArrow("SELECT (double_col + 1.23e1) / 2 - 2.5e1 FROM test");
+}
+
+TEST_F(CiderMathOpArrowTest, DecimalMathOpArrowTest) {
+  GTEST_SKIP_("decimal support for Arrow format is not ready.");
+  assertQueryArrow("SELECT integer_col + 0.123 FROM test");
+  assertQueryArrow("SELECT integer_col - 0.123 FROM test");
+  // modify output decimal type scale value of generated json file to pass dec_mul case.
+  assertQueryArrow("SELECT integer_col * 0.123 FROM test",
+                   "decimal_mul_output_scale_fixed.json");
+  // modify output decimal type scale value of generated json file to pass dec_div case.
+  assertQueryArrow("SELECT (integer_col + 0.8) / 2 FROM test",
+                   "decimal_div_output_scale_fixed.json");
+  assertQueryArrow("SELECT tinyint_col + 0.123 FROM test");
+  assertQueryArrow("SELECT smallint_col - 0.123 FROM test");
+  assertQueryArrow("SELECT double_col + 0.123 FROM test");
+  assertQueryArrow("SELECT double_col - 0.123 FROM test");
+  assertQueryArrow("SELECT double_col * 0.123 FROM test");
+  assertQueryArrow("SELECT double_col / 0.123 FROM test");
+  // modify type of bigint cast to decimal of generated json file to pass dec_int64 case.
+  assertQueryArrow("SELECT bigint_col + 0.123 FROM test",
+                   "decimal_bigint_cast_scale_fixed.json");
+}
+
+int main(int argc, char** argv) {
```

Comment:
Do we have any test case for error check case(like `ERR_OVERFLOW_OR_UNDERFLOW`)?

Reply:
Yes. `CiderMathOpArrowTest.ColumnMathOpBoundaryTest`.